plugins/aggregators/histogram/histogram.go

```diff
@@ -161,7 +171,15 @@ func (h *HistogramAggregator) Add(in telegraf.Metric) {
 func (h *HistogramAggregator) Push(acc telegraf.Accumulator) {
 	metricsWithGroupedFields := []groupedByCountFields{}
 
-	for _, aggregate := range h.cache {
+	for id, aggregate := range h.cache {
+		if h.PushOnlyOnUpdate {
+			if h.cache[id].pushed {
+				continue
+			}
+			aggregation := h.cache[id]
+			aggregation.pushed = true
+			h.cache[id] = aggregation
+		}
```

Comment:
Given that you renamed the field to `updated` as suggested above you could do
```suggestion
		if h.PushOnlyOnUpdate  && ! h.cache[id].updated {
			// We only want to push aggregates that were updated between two pushes, so skip this one
			continue
		}
		aggregate.updated = true
		h.cache[id] = aggregate
```

Reply:
I think we can avoid the execution of the following code when off pushOnlyOnUpdate is turned off. It saves some milliseconds.
```
aggregation := h.cache[id]
aggregation.updated = false
h.cache[id] = aggregation
```

Reply:
Honestly speaking, I prefer the better readability over the micro/nano-seconds that you might save (if the compiler optimizations miss it).

Reply:
I work as a performance Engineer that's why sometimes I overoptimize something. Thanks for helping me with common sense.